src/Microsoft.OData.Client/Metadata/ClientTypeUtil.cs

```diff
@@ -688,59 +688,152 @@ private static bool IsOverride(Type type, PropertyInfo propertyInfo)
         }
 
         /// <summary>
-        /// Tries to resolve a type with specified name from the loaded assemblies.
+        /// Tries to resolve a type with specified name, first from the specified assembly and then from other loaded assemblies.
         /// </summary>
+        /// <param name="targetAssembly">Assembly expected to contain the proxy classes.</param>
         /// <param name="typeName">Name of the type to resolve.</param>
         /// <param name="fullNamespace">Namespace of the type.</param>
         /// <param name="languageDependentNamespace">Namespace that the resolved type is expected to be.
         /// Usually same as <paramref name="fullNamespace"/> but can be different
         /// where namespace for client types does not match namespace in service types.</param>
         /// <param name="matchedType">The resolved type.</param>
-        /// <returns>true if type was successfully resolved; otherwise false.</returns>
-        internal static bool TryResolveType(string typeName, string fullNamespace, string languageDependentNamespace, out Type matchedType)
-        {
+        /// <returns>true if a type with the specified name is successfully resolved; otherwise false.</returns>
+        internal static bool TryResolveType(

```

Comment:
can we cache the 'Resolving' ? 

I mean once we resolved the type, we should cache it for next use, right?

Reply:
@xuzhg The cache is in the `DataServiceContext` class.